src/FindEmailAddress.java

```diff
@@ -0,0 +1,362 @@
+/*
+*
+* The project is for finding email address in the web site
+*
+* Create a command line program that will take an internet domain name (i.e. “jana.com”) and
+* print out a list of the email addresses that were found on that website only.
+* the application will find email addresses on any discoverable page of the website.
+*
+* We implemented a multithread method. each url request and email&href check is a threadTask
+*
+*
+*/
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Hashtable;
+import java.net.URI;
+import java.util.concurrent.*;
+
+/*
+*  This is the main class to get the email addresses
+*
+*/
+public class FindEmailAddress {
+
+    private String rootURL;
+    public String requestDomain;
+    public Hashtable<String,String> urlHistory;
+    public Hashtable<String,String> emails;
+    public static int maxUrlCheck = 200; //to avoid too many pages to check we add maximum number of url to check, it can be modified
```

Comment:
this seems like a very small number to me. what made you decide on 200 and not, say, 2000 or 20000?

Reply:
Can be any number, just for quick test and exam purpose. It even do not need any limitation. But it really took me very Long time to test such as "Boston.com"

Reply:
with your current implementation it does need to have some cap, I think, because your computer will definitely run out of memory before finishing, say, `twitter.com` or whatever.

I think a better solution is to print out the emails as you find them, instead of waiting until the entire site is scraped before printing them all out. then the script is useful whether or not it ever finishes, and it makes more sense to remove this cap.